runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -13611,7 +13611,7 @@ TR::Register *J9::Power::TreeEvaluator::arraycopyEvaluator(TR::Node *node, TR::C
    if (node->isForwardArrayCopy() && lengthNode->getOpCode().isLoadConst())
       {
       len = (lengthNode->getType().isInt32() ? lengthNode->getInt() : lengthNode->getLongInt());
-      if (len >= 0)
+      if (len >= 0 && len < 0x100000000LL)
```

Comment:
Is it worth adding a small comment why the upper range check is needed? Also will we handle the non-const case correctly?

Reply:
Added a comment in 6669a51adbd7ed48a5cb3ad681b52368855aa7af.

As for the non-const (helper call) case, it definitely generates code that doesn't trigger any asserts, but I'm honestly not sure whether it would actually function with such a long length if it were executed as I don't know that anybody has ever bothered to test it. The issue being fixed here definitely shouldn't affect it since it's a helper and not generated by the codegen at all. Generally, AFAIK it's not possible to actually execute an arraycopy node in OpenJ9 that has such a long length as they only arise from the optimizer and are completely unreachable in practice, e.g. as a result of #8530.